engine/common/rpc/convert/blocks_test.go

```diff
@@ -76,3 +78,146 @@ func TestConvertRootBlock(t *testing.T) {
 
 	assert.Equal(t, block.ID(), converted.ID())
 }
+
+// TestConvertBlockStatus tests converting protobuf BlockStatus messages to flow.BlockStatus.
+func TestConvertBlockStatus(t *testing.T) {
+	t.Parallel()
+
+	testCases := []struct {
+		name     string
+		pbStatus entities.BlockStatus
+		expected flow.BlockStatus
+	}{
+		{"Unknown", entities.BlockStatus_BLOCK_UNKNOWN, flow.BlockStatusUnknown},
+		{"Finalized", entities.BlockStatus_BLOCK_FINALIZED, flow.BlockStatusFinalized},
+		{"Sealed", entities.BlockStatus_BLOCK_SEALED, flow.BlockStatusSealed},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
```

Comment:
as of go1.22, for loops create new variables for each iteration, so you don't need this capture logic
https://go.dev/doc/go1.22#language
```suggestion
```